mount/mount_windows.go

```diff
@@ -97,32 +101,27 @@ func (mounter *Mounter) MountSensitive(source string, target string, fstype stri
 		getSMBMountMutex.LockKey(source)
 		defer getSMBMountMutex.UnlockKey(source)
 
-		var output string
-		var err error
 		username := allOptions[0]
 		password := allOptions[1]
-		if output, err = newSMBMapping(username, password, source); err != nil {
+		if output, err := newSMBMapping(username, password, source); err != nil {
 			klog.Warningf("SMB Mapping(%s) returned with error(%v), output(%s)", source, err, string(output))
 			if isSMBMappingExist(source) {
-				valid, errPath := isValidPath(source)
-				if errPath != nil {
-					return errPath
-				}
-				if valid {
-					err = nil
-					klog.V(2).Infof("SMB Mapping(%s) already exists and is still valid, skip error", source)
-				} else {
-					klog.V(2).Infof("SMB Mapping(%s) already exists while it's not valid, now begin to remove and remount", source)
+				valid, err := isValidPath(source)
+				if err == nil && !valid {
```

Comment:
I understand your logic, just thinking how to return the error in a more clear way. How about 
option1:
if isValid() return valid, move on,
if isValid() return error, call shouldRemovePath(error) to check whether the error indicates removing path is the only choice
optino2:
isValid return three value, (valid, shouldRemvoe, error) to clearly indicate the invalid path should be removed. 

What do you think? @andyzhangx @msau42

Reply:
thanks, PTAL.